File: lib/features/presentation/models/ten_pages_model.dart
class TenPages {
  String? dayId;
  String? bookTitle;
  int? pagesRead;
  String? summary; 
  bool? completed;


  TenPages({
    this.dayId,
    required this.bookTitle,
    required this.pagesRead,
    required this.summary,
    required this.completed,
  });

  // Factory constructor for creating a TenPages object from JSON
  factory TenPages.fromJson(Map<String, dynamic> json) {
    return TenPages(
      dayId: json['day_id'],
      bookTitle: json['book_title'],
      pagesRead: json['pages_read'],
      summary: json['summary'],
      completed: json['completed'],
    );
  }

  // Convert TenPages object to JSON  
  Map<String, dynamic> toJson() {
    return {
      'day_id': dayId,
      'pages_read': pagesRead,
      'book_title': bookTitle,
      'summary': summary,
      'completed': completed,
    };
  }
}

File: lib/backend/app/server/models/ten_pages_model.py
from beanie import Document
from typing import List, Optional
from beanie import PydanticObjectId


class TenPages(Document):
    class Settings:
        name = "TenPages"
    day_id: PydanticObjectId # Reference to the Day collection
    book_title: str
    pages_read: int
    summary: str
    completed: bool
    

# OLD VERSION (12/19)
# from beanie import Document
# from pydantic import EmailStr
# from typing import List, Optional
# from datetime import datetime

# class TenPages(Document):
#     class Settings:
#         name = "TenPages"
#     date : str  # Date of the day
#     firebase_uid : str
#     completed: bool
#     summary: str
#     bookTitle: str
#     completed: bool
#     pagesRead: int







